### PR TITLE
[spirv] Improve SPIR-V structures and tests for them

### PR DESCRIPTION
**Pull request by antiagainst**
* Changed Instruction to a class
* Created SimpleInstBuilder for building instructions in tests
* Various cosmetic changes

### CODE REVIEW

File: tools/clang/include/clang/SPIRV/Structure.h [hunk 1]
```diff
@@ -35,30 +35,66 @@
...
+  // Move constructor/assignment
   BasicBlock(BasicBlock &&that);
   BasicBlock &operator=(BasicBlock &&that);
 
   /// \brief Returns true if this basic block is empty, which has no <label-id>
   /// assigned and no instructions.
   inline bool isEmpty() const;
-  /// \brief Clears all instructions in this basic block and turns this basic
-  /// block into an empty basic block.
+  /// \brief Clears the everything in this basic block and turns it into an
```
Comment (ehsannas) [inline tools/clang/include/clang/SPIRV/Structure.h]: s/the everything/everything

File: tools/clang/lib/SPIRV/InstBuilderAuto.cpp [hunk 2]
```diff
@@ -16,7 +16,7 @@
 namespace clang {
 namespace spirv {
 
-static_assert(spv::Version == 0x00010000 && spv::Revision == 10,
+static_assert(spv::Version == 0x00010000 && spv::Revision == 11,
```
Comment (ehsannas) [inline tools/clang/lib/SPIRV/InstBuilderAuto.cpp]: why did the revision change?

Comment (antiagainst) [inline tools/clang/lib/SPIRV/InstBuilderAuto.cpp]: SPIRV-Headers was updated three days ago: https://github.com/KhronosGroup/SPIRV-Headers/commit/63e1062a194750b354d48be8c16750d7a4d0dc4e.

File: tools/clang/include/clang/SPIRV/Structure.h (hunk 1 above)
Comment (antiagainst) [inline tools/clang/include/clang/SPIRV/Structure.h]: Done

**Pull request merged**